web-admin/src/features/organizations/settings/UploadImagePopover.svelte

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+  import { createAdminServiceCreateAsset } from "@rilldata/web-admin/client";
+  import { CANONICAL_ADMIN_URL } from "@rilldata/web-admin/client/http-client";
+  import {
+    Popover,
+    PopoverContent,
+    PopoverTrigger,
+  } from "@rilldata/web-common/components/popover/index.js";
+  import { Button } from "@rilldata/web-common/components/button/index.js";
+  import FileInput from "@rilldata/web-common/components/forms/FileInput.svelte";
+  import EditIcon from "@rilldata/web-common/components/icons/EditIcon.svelte";
+  import { extractFileExtension } from "@rilldata/web-common/features/entity-management/file-path-utils";
+  import { getAttrs, builderActions } from "bits-ui";
+
+  export let imageUrl: string;
+  export let accept: string;
+  export let label: string;
+  export let organization: string;
+  export let loading: boolean;
+  export let error: string;
+  export let onSave: (assetId: string) => Promise<void>;
+  export let onRemove: () => Promise<void>;
+
+  $: url = imageUrl;
+
+  let open = false;
+  let assetId = "";
+
+  const assetCreator = createAdminServiceCreateAsset();
+
+  async function uploadFile(file: File) {
+    const ext = extractFileExtension(file.name);
+    const assetResp = await $assetCreator.mutateAsync({
+      organizationName: organization,
+      data: {
+        type: "image",
+        name: label,
+        extension: ext,
+        public: true,
+        estimatedSizeBytes: file.size.toString(),
+      },
+    });
+
+    await fetch(assetResp.signedUrl, {
+      method: "PUT",
+      body: file,
+      headers: assetResp.signingHeaders,
+    });
+    assetId = assetResp.assetId;
+    return `${CANONICAL_ADMIN_URL}/v1/assets/${assetId}/download`;
+  }
+
+  function onCancel() {
+    open = false;
+    url = imageUrl;
```

Comment:
Just checking – is this needed? I see the reactive statement above (`$: url = imageUrl;`). Regardless, it's probably worth clarifying with a comment why the `url` variable is needed in addition to `imageUrl`.

Reply:
Adding a comment to clarify why we need it.